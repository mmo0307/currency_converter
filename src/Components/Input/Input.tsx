import {IProps} from "../../interface/Iprops.input";
import React from "react";
//import {nanoid} from "nanoid";

export const Input = (
    {
        text,
        value,
        currencyArray = ['UAH'],
        handleValue,
        setSelectCurrency,
        selectCurrency,
        handleChangeSelect
    }:IProps
):JSX.Element => {
    return (
        <div>
            <label htmlFor="price" className="block text-sm font-medium text-gray-700">
                {text}
            </label>
            <div className="relative mt-1 rounded-md shadow-sm">
                <input
                    min={0}
                    max={1000000}
                    type="number"
                    name="price"
                    id="price"
                    className="block w-full rounded-md border-gray-300 pl-4 pr-[70px] focus:border-indigo-500 focus:ring-indigo-500 sm:text-sm"
                    placeholder="0.00"
                    value={value?.value}
                    onChange={event => handleValue?.(event.target.value)}
                />
                <div className="absolute inset-y-0 right-0 flex items-center">
                    <select
                        id="currency"
                        name="currency"
                        className="h-full rounded-md border-transparent bg-transparent py-0 pl-2 pr-7 text-gray-500 focus:border-indigo-500 focus:ring-indigo-500 sm:text-sm"
                        value={selectCurrency}
                        onChange={(event) => {
                            setSelectCurrency?.((prev) => ({...prev, currency: event.target.value}));
                            handleChangeSelect?.(event.target.value);
                        }}
                    >
                        {currencyArray.map(item =>
                            <option
                                key={(Math.random()).toString()}
                                //key={nanoid()}
                            >{item}</option>)}
                    </select>
                </div>
            </div>
        </div>
    )
}
