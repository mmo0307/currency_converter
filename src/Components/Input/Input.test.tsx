import {render} from "@testing-library/react";
import React from "react";
import {Input} from "./Input";

const currencyArray = ['UAH', 'USD', 'EUR'];
describe('Input renders', () => {
    it('Input renders without someone parameters', () => {
        render(<Input text="TEXT" currencyArray={currencyArray}/>)
    });

    it('Input snapshot without someone parameters', () => {
        expect(
            render(<Input text="TEXT" currencyArray={currencyArray} />)
        ).toMatchSnapshot();
    })
})