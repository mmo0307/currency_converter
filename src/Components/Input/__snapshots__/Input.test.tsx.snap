// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders Input snapshot without someone parameters 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <label
          class="block text-sm font-medium text-gray-700"
          for="price"
        >
          TEXT
        </label>
        <div
          class="relative mt-1 rounded-md shadow-sm"
        >
          <input
            class="block w-full rounded-md border-gray-300 pl-4 pr-[70px] focus:border-indigo-500 focus:ring-indigo-500 sm:text-sm"
            id="price"
            max="1000000"
            min="0"
            name="price"
            placeholder="0.00"
            type="number"
            value=""
          />
          <div
            class="absolute inset-y-0 right-0 flex items-center"
          >
            <select
              class="h-full rounded-md border-transparent bg-transparent py-0 pl-2 pr-7 text-gray-500 focus:border-indigo-500 focus:ring-indigo-500 sm:text-sm"
              id="currency"
              name="currency"
            >
              <option>
                UAH
              </option>
              <option>
                USD
              </option>
              <option>
                EUR
              </option>
            </select>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <label
        class="block text-sm font-medium text-gray-700"
        for="price"
      >
        TEXT
      </label>
      <div
        class="relative mt-1 rounded-md shadow-sm"
      >
        <input
          class="block w-full rounded-md border-gray-300 pl-4 pr-[70px] focus:border-indigo-500 focus:ring-indigo-500 sm:text-sm"
          id="price"
          max="1000000"
          min="0"
          name="price"
          placeholder="0.00"
          type="number"
          value=""
        />
        <div
          class="absolute inset-y-0 right-0 flex items-center"
        >
          <select
            class="h-full rounded-md border-transparent bg-transparent py-0 pl-2 pr-7 text-gray-500 focus:border-indigo-500 focus:ring-indigo-500 sm:text-sm"
            id="currency"
            name="currency"
          >
            <option>
              UAH
            </option>
            <option>
              USD
            </option>
            <option>
              EUR
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
